src/main/java/com/minecolonies/coremod/colony/requestsystem/management/manager/StandardRequestManager.java

```diff
@@ -56,94 +46,72 @@
 
-    private static final String NBT_TOKEN       = "Token";
-    private static final String NBT_ASSIGNMENTS = "Assignments";
+    private IToken<?> requestIdentitiesDataStoreId;
 
-    private static final String NBT_REQUEST = "Request";
-    ////---------------------------NBTTags-------------------------\\\\
+    private IToken<?> requestResolverIdentitiesDataStoreId;
 
-    /**
-     * BiMap that holds unique token to provider lookup.
-     */
-    @NotNull
-    private final BiMap<IToken<?>, IRequestResolverProvider> providerBiMap = HashBiMap.create();
+    private IToken<?> providerRequestResolverAssignmentDataStoreId;
 
-    /**
-     * BiMap that holds unique token to resolver lookup.
-     */
-    @NotNull
-    private final BiMap<IToken<?>, IRequestResolver<? extends IRequestable>> resolverBiMap = HashBiMap.create();
+    private IToken<?> requestResolverRequestAssignmentDataStoreId;
 
-    /**
-     * BiMap that holds unique token to request lookup.
-     */
-    @NotNull
-    private final BiMap<IToken<?>, IRequest<?>> requestBiMap = HashBiMap.create();
+    private IToken<?> requestableTypeRequestResolverAssignmentDataStoreId;
 
-    /**
-     * Map that holds the resolvers that are linked to a given provider.
-     */
-    @NotNull
-    private final Map<IToken<?>, ImmutableCollection<IToken<?>>> providerResolverMap = new HashMap<>();
+    private IToken<?> playerRequestResolverId;
 
-    /**
-     * Map that holds the requests that are linked to a given resolver.
-     */
-    @NotNull
-    private final Map<IToken<?>, Set<IToken<?>>> resolverRequestMap = new HashMap<>();
+    private IToken<?> retryingRequestResolverId;
 
-    /**
-     * Map that holds the resolver that is linked to a given request.
-     */
-    @NotNull
-    private final Map<IToken<?>, IToken<?>> requestResolverMap = new HashMap<>();
+    private IDataStoreManager dataStoreManager;
 
-    /**
-     * Map that holds the class that resolver can resolve. Used during lookup.
-     */
-    @NotNull
-    private final Map<TypeToken<?>, Collection<IRequestResolver<?>>> requestClassResolverMap = new HashMap<>();
     /**
      * Colony of the manager.
      */
     @NotNull
     private final IColony colony;
-    /**
-     * The fallback resolver used to resolve directly to the player.
-     */
-    @NotNull
-    private       IPlayerRequestResolver                       playerResolver          = null;
-    /**
-     * The fallback resolver used to resolve using retries.
-     * Not all requests might support this feature, requests that do should implement {@link IRetryable} on their requestable.
-     * Anything that implements {@link IDeliverable} is by definition retryable.
-     */
+
     @NotNull
-    private IRetryingRequestResolver retryingResolver = null;
+    private int version = -1;
 
     public StandardRequestManager(final IColony colony)
     {
         this.colony = colony;
+        reset();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove this call from a constructor to the overridable "reset" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)